build/lib/pydatastructs/miscellaneous_data_structures/stack.py

```diff
@@ -0,0 +1,189 @@
+                kwargs.get('dtype', int))
+
+        elif implementation == "ll":
+            return Linked_Stacks()
+
+        raise NotImplementedError(
+                "%s hasn't been implemented yet."%(implementation))
+
+
+    def push(self, *args, **kwargs):
+        raise NotImplementedError(
+            "This is an abstract method.")
+
+    def pop(self, *args, **kwargs):
+        raise NotImplementedError(
+              "This is an abstract method.")
+
+    @property
+    def is_empty(self):
+        return None
+
+    @property
+    def peek(self):
+        return None
+
+
+class ArrayStack(Stack):
+
+    __slots__ = ['items']
+
+    def __new__(cls, items=None, dtype=NoneType):
+        if items is None:
+            items = DynamicOneDimensionalArray(dtype, 0)
+        else:
+            items = DynamicOneDimensionalArray(dtype, items)
+        obj = object.__new__(cls)
+        obj.items = items
+        return obj
+
+    def push(self, x):
+        if self.is_empty:
+            self.items._dtype = type(x)
+        self.items.append(x)
+
+    def pop(self):
+        if self.is_empty:
+            raise ValueError("Stack is empty")
+
+        top_element = dc(self.items[self.items._last_pos_filled])
+        self.items.delete(self.items._last_pos_filled)
+        return top_element
+
+    @property
+    def is_empty(self):
+        return self.items._last_pos_filled == -1
+
+    @property
+    def peek(self):
+        return self.items[self.items._last_pos_filled]
+
+    def __str__(self):
+        """
+        Used for printing.
+        """
+        return str(self.items._data)
+
+class Linked_Stacks(Stack):
+
+    """Representation of Stack Data Structure using Doubly Linked List
+    Methods
+    ===========
+    push :
+    A normal push operation to the Stack
+
+    pop :
+    Delete the top most element from the stack
+    Returns the value of top element
+
+    peek :
+    returns the value of top element
+
+    is_empty :
+    Checks for whether a Stack is been empty or not
+    Return True if empty else False
+
+    __str__ :
+    Used for Printing the Stack
+
+    """
+    __slots__ = ["dll"]
+    def __new__(cls):
+        dll = DoublyLinkedList()
+        obj = object.__new__(cls)
+        obj.dll = dll
+        obj.top = dll.head
+        return obj
+
+
+    def push(self,data):
+        self.dll.append(data)
+        if self.top is None:
+            self.top = self.dll.head
+        self.top = self.top.next
+
+    def pop(self):
+        if not self.is_empty:
+            self.data = self.top.data
+            self.top = self.top.prev
+            return self.data
+        else:
+            raise ValueError
+
+    @property
+    def is_empty(self):
+        if self.top is None:
+            return 1
+        return 0
+        return self.top is None
+
+
+    @property
+    def peek(self):
+        if self.top is not None:
+            return self.top.data
+        raise ValueError
```

Comment:
ValueError takes a string argument. For your reference: https://kite.com/python/answers/how-to-raise-a-valueerror-in-python